File: app/Models/Booking.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;

class Booking extends Model
{
    use HasFactory;

    protected $fillable = [
        'code',
        'booking_date',
        'passenger_count',
        'client',
        'travel_package_id',
        'package_name',
        'package_code',
        'entry_type',
        'destination',
        'gross_revenue',
        'payment_status',
        'booking_status',
        'staff_assigned',
        'source',
        'notes',
    ];

    protected function casts(): array
    {
        return [
            'booking_date' => 'date',
            'passenger_count' => 'integer',
            'gross_revenue' => 'integer',
        ];
    }

    public function travelPackage(): BelongsTo
    {
        return $this->belongsTo(TravelPackage::class);
    }
}

File: dotnet/ProphetOps.Domain/Booking.cs
namespace ProphetOps.Domain;

public class Booking
{
    public int Id { get; set; }
    public string Code { get; set; } = "";
    public DateOnly BookingDate { get; set; }
    public int PassengerCount { get; set; }
    public string Client { get; set; } = "";
    public int? TravelPackageId { get; set; }
    public TravelPackage? TravelPackage { get; set; }
    public string PackageName { get; set; } = "";
    public string? PackageCode { get; set; }
    public string EntryType { get; set; } = "Package preset";
    public string Destination { get; set; } = "";
    public int GrossRevenue { get; set; }
    public string PaymentStatus { get; set; } = "Pending";
    public string BookingStatus { get; set; } = "Pending";
    public string? StaffAssigned { get; set; }
    public string Source { get; set; } = "Manual quotation";
    public string? Notes { get; set; }

    public DateTime? VoidedAt { get; set; }
    public string? VoidedBy { get; set; }
    public string? VoidReason { get; set; }

    public bool IsVoided => VoidedAt is not null;
}
